include/SDL3/SDL_main.h

```diff
@@ -89,6 +89,14 @@
         /* We need to export SDL_main so it can be launched from Java */
         #define SDLMAIN_DECLSPEC    SDL_DECLSPEC
 
+        /* As this is launched from Java, the real entry point (main() function)
+           is outside of the the binary built from this code.
+           This define makes sure that, unlike on other platforms, SDL_main.h
+           and SDL_main_impl.h only provide an `SDL_main()` function (to be called
+           from Java), but no native `int main(int argc, char* argv[])` or similar.
+         */
+        #define SDL_MAIN_OUTSIDE_MAIN
```

Comment:
```suggestion
        #define SDL_MAIN_EXPORTED
```
What about this?

Reply:
not sure..
or maybe something like SDL_MAIN_EXTERNAL or SDL_MAIN_IS_EXTERNAL?

Reply:
OTOH, maybe I like EXPORTED better after all, EXTERNAL sounds too much like something that abstracts `extern "C"` or similar

Reply:
what do you think about moving `#define SDLMAIN_DECLSPEC    SDL_DECLSPEC`
to the place where currently `#ifndef SDL_DECLSPEC #define SDL_DECLSPEC #endif` is, as
```c
#ifdef SDL_MAIN_EXPORTED
#define SDLMAIN_DECLSPEC  SDL_DECLSPEC
#else
#define SDLMAIN_DECLSPEC
#endif
```

Reply:
I like it!